src/pip/_internal/self_outdated_check.py

```diff
@@ -233,7 +233,7 @@ def pip_self_version_check(session, options):
             "You are using pip version %s; however, version %s is "
             "available.\nYou should consider upgrading via the "
             "'%s install --upgrade pip' command.",
-            pip_version, pypi_version, pip_cmd
+            str(pip_version), pypi_version, pip_cmd
```

Comment:
Is this needed? I thought the `%s` format tag would do this automatically.

Reply:
It's not actually. I was having issues with asserting tests because `pip_version` is of `Version` type. But I think now with using `caplog`, we won't need this change.